### ISSUE TITLE
ExecutionTimeOf is very limited

### ISSUE THREAD

**Issue by PhilipDaniels** [fluentassertions/fluentassertions#292]
Using `ExecutionTimeOf` is very limited, there is basically only `ExecutionTimeAssertions.ShouldNotExceed` available. I would like to have most of the `SimpleTimeSpanAssertions` available. Namely

```
BeCloseTo
BeGreatorOrEqualTo
BeGreaterThan
BeLessOrEqualTo
BeLessThan
```

If we had these it would be possible to get rid of `ShouldNotExceed`, because it is semantically equivalent to `BeLessOrEqualTo`. 

A simple case of hoisting some of the `SimpleTimeSpanAssertions` into a new base class?

**Comment by jnyrup** [comment]
@dennisdoomen FYI, `ShouldNotExceed` is the only function left in release-5.0 branch, that doesn't align with the `Should().` pattern.

How about the API was changed, so the two examples from the documentation
```c#
subject.ExecutionTimeOf(s => s.ExpensiveMethod()).ShouldNotExceed(500.Milliseconds());
someAction.ExecutionTime().ShouldNotExceed(100.Milliseconds());
```

would become
´´´c#
subject.ExecutionTimeOf(s => s.ExpensiveMethod()).Should().BeLessOrEqualTo(500.Milliseconds());
someAction.ExecutionTime().Should().BeLessOrEqualTo(100.Milliseconds());
´´´

**Comment by dennisdoomen** [comment]
Didn't even know that existed. I just tweeted that beta 3 is the last beta with breaking changes 😢

**Comment by jnyrup** [comment]
This is fixed on #653
